code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/ProjectResourceManager.java

```diff
@@ -0,0 +1,135 @@
+import io.cattle.platform.json.JsonMapper;
+import io.github.ibuildthecloud.gdapi.context.ApiContext;
+import io.github.ibuildthecloud.gdapi.exception.ClientVisibleException;
+import io.github.ibuildthecloud.gdapi.factory.SchemaFactory;
+import io.github.ibuildthecloud.gdapi.model.ListOptions;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.util.ResponseCodes;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class ProjectResourceManager extends AbstractJooqResourceManager {
+
+    private static final String GITHUB_EXTERNAL_TYPE = "github";
+    private static final String TEAM_SCOPE = "project:github_team";
+    private static final String ORG_SCOPE = "project:github_org";
+    private static final String USER_SCOPE = "project:github_user";
+    private static final String AUTH = "Authorization";
+
+    JsonMapper jsonMapper;
+    GithubUtils githubUtils;
+    GithubClient githubClient;
+    AuthDao authDao;
+
+    @Override
+    public String[] getTypes() {
+        return new String[] { "project" };
+    }
+
+    @Override
+    public Class<?>[] getTypeClasses() {
+        return new Class[] {};
+    }
+
+    @Override
+    public Object listInternal(SchemaFactory schemaFactory, String type, Map<Object, Object> criteria, ListOptions options) {
+        HttpServletRequest request = ApiContext.getContext().getApiRequest().getServletContext().getRequest();
+        String token = request.getHeader(AUTH);
+        if (StringUtils.isEmpty(token)) {
+            return null;
+        }
+        List<String> teamIds = githubUtils.validateAndFetchTeamIdsFromToken(token);
+        List<String> orgIds = githubUtils.validateAndFetchOrgIdsFromToken(token);
+        String userId = githubUtils.validateAndFetchAccountIdFromToken(token);
+        Account userAccount = authDao.getAccountByExternalId(userId, GITHUB_EXTERNAL_TYPE);
+        if (userAccount == null) {
+            return null;
+        }
+        return authDao.getAccessibleProjects(userAccount.getId(), orgIds, teamIds);
+    }
+
+    @Override
+    protected Object createInternal(String type, ApiRequest request) {
+        if (!AccountConstants.PROJECT.equals(type)) {
+            return null;
+        }
+
+        return createProject(request);
+    }
+
+    private Account createProject(ApiRequest apiRequest) {
+        HttpServletRequest request = apiRequest.getServletContext().getRequest();
+        String token = request.getHeader(AUTH);
+        if (StringUtils.isEmpty(token)) {
+            throw new ClientVisibleException(ResponseCodes.FORBIDDEN);
+        }
+        @SuppressWarnings("unchecked")
+        Map<String, Object> account = jsonMapper.convertValue(apiRequest.getRequestObject(), Map.class);
```

Comment:
Use `Map<String,Object> stuff = CollectionUtils.toMap(request.getRequestObject());`  This is how other code does it.